crates/sui-graphql-rpc/src/types/event.rs

```diff
@@ -71,17 +72,21 @@ impl Event {
             .extend()
     }
 
-    /// Addresses of the senders of the event
-    async fn senders(&self) -> Result<Option<Vec<Address>>> {
-        let mut addrs = Vec::with_capacity(self.stored.senders.len());
-        for sender in &self.stored.senders {
-            let Some(sender) = &sender else { continue };
-            let address = SuiAddress::from_bytes(sender)
-                .map_err(|e| Error::Internal(format!("Failed to deserialize address: {e}")))
-                .extend()?;
-            addrs.push(Address { address });
+    /// Address of the sender of the event
+    async fn sender(&self) -> Result<Option<Address>> {
+        let Some(Some(sender)) = self.stored.senders.first() else {
+            return Ok(None);
+        };
+
+        let address = SuiAddress::from_bytes(sender)
+            .map_err(|e| Error::Internal(format!("Failed to deserialize address: {e}")))
+            .extend()?;
+
+        if address.as_slice() != NativeSuiAddress::ZERO.as_ref() {
+            return Ok(Some(Address { address }));
         }
-        Ok(Some(addrs))
+
+        Ok(None)
     }
```

Comment:
nit: slight preference to early return with the error and end with the happy path, (for consistency with the earlier check), but not a big deal.